crates/lib3h/src/dht/dht_protocol.rs

```diff
@@ -2,29 +2,109 @@ use crate::dht::PeerAddress;
+    DhtRequestToChildResponse,
+    Lib3hError,
+>;
+pub type DhtEndpoint = GhostEndpoint<
+    DhtRequestToChild,
+    DhtRequestToChildResponse,
+    DhtRequestToParent,
+    DhtRequestToParentResponse,
+    Lib3hError,
+>;
+pub type ChildDhtWrapperDyn<UserData> = GhostParentWrapperDyn<
+    UserData,
+    DhtContext,
+    DhtRequestToParent,
+    DhtRequestToParentResponse,
+    DhtRequestToChild,
+    DhtRequestToChildResponse,
+    Lib3hError,
+>;
+
+pub type DhtToChildMessage =
+    GhostMessage<DhtRequestToChild, DhtRequestToParent, DhtRequestToChildResponse, Lib3hError>;
+
+pub type DhtToParentMessage =
+    GhostMessage<DhtRequestToParent, DhtRequestToChild, DhtRequestToParentResponse, Lib3hError>;
+
+#[derive(Debug)]
+pub enum DhtContext {
+    NoOp,
+    RequestAspectsOf {
+        entry_address: Address,
+        aspect_address_list: Vec<Address>,
+        msg: EntryListData,
+        request_id: String,
+    },
+    RequestEntry(DhtToChildMessage),
+    QueryEntry(QueryEntryData),
+}
+
+#[derive(Debug)]
+pub enum DhtRequestToChild {
+    /// Commands
+    /// Parent received a gossip bundle from a remote peer, and asks us to handle it.
     HandleGossip(RemoteGossipBundleData),
-    /// Owner wants a specific entry.
-    FetchEntry(FetchDhtEntryData),
-    /// Owner wants us to hold a peer discovery data item.
+    /// Parent wants us to hold a peer discovery data item.
     HoldPeer(PeerData),
-    /// Owner notifies us that it is holding one or several Aspects for an Entry.
+    /// Parent notifies us that it is holding one or several Aspects for an Entry.
     /// Note: Need an EntryData to know the aspect addresses, but aspects' content can be empty.
     HoldEntryAspectAddress(EntryData),
-    /// Owner wants us to bookkeep an entry and broadcast it to neighbors
+    /// Parent wants us to bookkeep an entry and broadcast it to neighbors
     BroadcastEntry(EntryData),
-    /// Owner notifies us that is is not holding an entry anymore.
+    /// Parent notifies us that is is not holding an entry anymore.
     DropEntryAddress(Address),
-    /// Owner's response to ProvideEntry request
-    EntryDataResponse(FetchDhtEntryResponseData),
+
+    /// Requests
+    /// Parent wants PeerData for a specific Peer
+    RequestPeer(String),
+    /// Parent wants the list of peers we are holding
+    RequestPeerList,
+    /// Parent wants PeerData of this entity
+    RequestThisPeer,
+    /// Parent wants the list of entries we are holding
+    RequestEntryAddressList,
+    /// Parent wants address' we have for an entry
+    RequestAspectsOf(Address),
+    /// Parent wants a specific entry.
+    RequestEntry(Address),
 }
 
-#[derive(Debug, PartialEq, Clone)]
-pub enum DhtEvent {
+#[derive(Debug)]
+pub enum DhtRequestToChildResponse {
+    RequestPeer(Option<PeerData>),
+    RequestPeerList(Vec<PeerData>),
+    RequestThisPeer(PeerData),
+    RequestEntryAddressList(Vec<Address>),
+    RequestAspectsOf(Option<Vec<Address>>),
```

Comment:
What is difference between `None` and `Some(vec![])` here?

Reply:
`None` means there is no entry at all.
`Some(vec![]` means there is an entry with no aspects.